File: react_web/src/api/export.ts
import { api } from "./client";

export interface ExportSnapshotResponse {
  ok: boolean;
  directory: string;
  files: string[];
}

export async function postExportSnapshot(): Promise<ExportSnapshotResponse> {
  const { data } = await api.post<ExportSnapshotResponse>("/export/snapshot");
  return data;
}

File: api/routes/export.py
from fastapi import APIRouter, Depends

from dependencies import get_current_user
from services.file_export import DATA_DIR, export_snapshot_safe

router = APIRouter(
    prefix="/export",
    tags=["Export"],
    dependencies=[Depends(get_current_user)],
)


@router.post("/snapshot")
def write_csv_snapshot():
    """Regenerate ``data/orders.csv``, ``data/agvs.csv``, and ``data/logs.csv``."""
    export_snapshot_safe()
    return {
        "ok": True,
        "directory": str(DATA_DIR.resolve()),
        "files": ["orders.csv", "agvs.csv", "logs.csv", "warehouse_audit.log"],
    }
